### PR TITLE
Update GKE Metadata Google Cloud Client Defaults

### PR DESCRIPTION
**Pull request by mbolt35**
## What does this PR change?
* The metadata client was hanging indefinitely in some cases running locally (where the endpoint was unreachable). Updating the HTTPClient used to match the default provided in the google cloud API: https://github.com/googleapis/google-cloud-go/blob/4c6bb3cb062b3e208a617462ee4c4c64660ea17f/compute/metadata/metadata.go#L66-L77
  * This also stopped the indefinite hang.

### CODE REVIEW

File: pkg/util/httputil/roundtrip.go [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-	base      http.RoundTripper
 }
 
 // NewUserAgentTransport creates a RoundTripper that attaches the configured user agent.
 func NewUserAgentTransport(userAgent string, base http.RoundTripper) http.RoundTripper {
 	return &userAgentTransport{
-		userAgent: userAgent,
-		base:      base,
+		RoundTripper: base,
+		userAgent:    userAgent,
```
Comment (michaelmdresser) [inline pkg/util/httputil/roundtrip.go]: What's going on with this `base: base` to `RoundTripper: base` change?

**Review by michaelmdresser — APPROVED** [review]

**Review by AjayTripathy — APPROVED** [review]

File: pkg/util/httputil/roundtrip.go (hunk 1 above)
Comment (mbolt35) [reply pkg/util/httputil/roundtrip.go]: It makes the base the actual base. When I made this change originally, I was moving most of the construction inside this type, but given that this just overrides the roundtripper, it made more sense just to move the HTTPClient defaults back out to the provider. This is what happened when I put it back together, and the way that it existed when I tested.

Comment (mbolt35) [reply pkg/util/httputil/roundtrip.go]: Semantically identical, I'd be glad to change it back if you want to roll the dice :)

Comment (michaelmdresser) [reply pkg/util/httputil/roundtrip.go]: Huh. You're definitely the expert here, and I trust that expertise.

Comment (mbolt35) [reply pkg/util/httputil/roundtrip.go]: No, you're right - I just sloppily threw it back together. The tests do pass, but I won't chance it. Will need to test live again, but I'll do that before merging.

**Pull request merged**